presto-spi/src/main/java/io/prestosql/spi/block/DictionaryBlock.java

```diff
@@ -93,6 +93,9 @@ public DictionaryBlock(int idsOffset, int positionCount, Block dictionary, int[]
         this.retainedSizeInBytes = INSTANCE_SIZE + dictionary.getRetainedSizeInBytes() + sizeOf(ids);
 
         if (dictionaryIsCompacted) {
+            if (dictionary instanceof DictionaryBlock) {
+                throw new IllegalArgumentException("compacted dictionary should not have dictionary base block");
```

Comment:
`checkArgument(!(dictionary instanceof DictionaryBlock), ...`

Reply:
No guava in SPI